tensorflow/lite/java/src/main/java/org/tensorflow/lite/Tensor.java

```diff
@@ -420,7 +420,8 @@ private void throwIfShapeIsIncompatible(Object o) {
       // Note that we allow the client to provide a ByteBuffer even for non-byte Tensors.
       // In such cases, we only care that the raw byte capacity matches the tensor byte capacity.
       int oBytes = isByteBuffer(o) ? oBuffer.capacity() : oBuffer.capacity() * dtype.byteSize();
-      if (bytes != oBytes) {
+      // Allow dynamic output sizes, but be sure to allocate enough bytes in the destination
+      if (bytes > oBytes) {
```

Comment:
What about the case where we're copying *from* an input buffer that is too large? I think we might need to distinguish between the cases where we're copying to/from an buffer?

Reply:
Is this not the original idea of this check and throwing this exception (which will still work for:
"_copying from an input buffer that is too large_")? 

I needed exactly this specific change as I don't know before the expected dynamic returned buffer size from the decoded characters found by ctc_beam_search_decoder. And in Java a buffer fits always in an _equal or larger_ output buffer, so I pre-allocated the result buffer always larger than the expected results can be.

Reply:
There are 4 cases we need to consider:

1) The input (user) buffer is too large for the input *tensor*. - This should be an error.
2) The input buffer is too small for the input tensor - This should probably also be an error.
3) The output (user) buffer is too large for the output tensor - This is probably fine and is what you want to handle.
4) The output buffer is too small for the output tensor - This should be an error.

This method gets called for all 4 cases, but isn't reporting an error for (1).

Reply:
Yes, true, you're right, checked it, it is used for both and as my input images were always right for my model, it worked... What do you recommend, copy the method and allow it only for _copyTo()_?